aurt/robot_dynamics.py

```diff
@@ -102,24 +95,20 @@ def compute_regressor_joint():
                 reg_j = sp.Matrix.hstack(reg_j, reg_j_rbd_par_j, reg_j_jd)
             return reg_j
 
-        return compute_regressor_joint()#cache_object(self.filepath_regressor_joint(j+1), compute_regressor_joint)
+        return compute_regressor_joint()
 
-    def regressor(self, output_filename="robot_dynamics_regressor"):
-        filepath_regressor = from_cache(output_filename)
+    def regressor(self, output_filename="robot_dynamics_regressor.pickle"):
+        return self._cache.get_or_cache(output_filename, self.compute_regressor)
 
-        def compute_regressor():

```

Comment:
why is this function not to be defined inside the scope of "regressor()". It is only used inside this function.